csm/core/services/stats.py

```diff
@@ -131,7 +131,7 @@ async def get_metrics(self, stats_id, metrics_list, from_t, to_t, interval,
                 else:
                     panels[li[0]]["unit"].append(li[2])
         except:
-            raise CsmInternalError("Invalid metric list for Stats %s" %metrics_list)
+            raise InvalidRequest("Invalid metric list for Stats %s" %metrics_list)
```

Comment:
Catch Invalid request

Reply:
@pawankumarsrivastava, InvalidRequest is the CSM exception that is mapped to 400 error code. It's purpose in services is to be raised, not caught. Here and in the snippet above that you've commented the code handles inputs from the request and converts any error occured in process to the InvalidRequest. That's my understanding. Thus I'm OK with these edits.
@SrikantVishwakarma, please, also post your assumptions here.

Reply:
@avrnv May be my comment was not clear. What I am suggesting is to catch specific expection, in this case it maybe OutofBounds. Don't use except alone.

Reply:
> caught

I am agree with @avrnv, CSM exceptions are mapped to specific  error codes, and it's purpose is to be raised not caught.

Reply:
Addressed  review comments